File: 2022-2023/sourcecode/lab/lab13/DaggerHiltROOMBasicsJava/app/src/main/java/pl/edu/uwr/pum/dagger_hilt_room_basicsjava/data/db/AppDao.java
package pl.edu.uwr.pum.dagger_hilt_room_basicsjava.data.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import pl.edu.uwr.pum.dagger_hilt_room_basicsjava.data.Student;

@Dao
public interface AppDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void addItem(Student student);

    @Query("SELECT * FROM student")
    LiveData<List<Student>> readAllData();
}

File: 2022-2023/sourcecode/lab/lab13/DaggerHiltROOMBasicsKotlin/app/src/main/java/pl/edu/uwr/pum/dagger_hilt_room_basicskotlin/data/db/AppDao.kt
package pl.edu.uwr.pum.dagger_hilt_room_basicskotlin.data.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import pl.edu.uwr.pum.dagger_hilt_room_basicskotlin.data.Student

@Dao
interface AppDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun addItem(student: Student)

    @Query("SELECT * FROM student")
    fun readAllData(): LiveData<List<Student>>
}